### PR TITLE
add generate mask target op

### PR DESCRIPTION
**Pull request by jerrywgz**

### CODE REVIEW

File: python/paddle/fluid/tests/unittests/test_generate_mask_target_op.py [hunk 1]
```diff
@@ -0,0 +1,267 @@
...
+
+    def make_rois(self):
+        rois = []
+        self.rois_lod = [[]]
+        for bno in range(self.batch_size):
+            self.rois_lod[0].append(bno + 1)
+            for i in range(bno + 1):
+                xywh = np.random.rand(4)
+                xy1 = xywh[0:2] * 2
+                wh = xywh[2:4] * (16 - xy1)
```
Comment (xingyuanbu) [inline python/paddle/fluid/tests/unittests/test_generate_mask_target_op.py]: Don't hard write 16 for image size

File: python/paddle/fluid/tests/unittests/test_generate_mask_target_op.py [hunk 2]
```diff
@@ -0,0 +1,267 @@
...
+                xy2 = xy1 + wh
+                roi = [bno, xy1[0], xy1[1], xy2[0], xy2[1]]
```
Comment (xingyuanbu) [inline python/paddle/fluid/tests/unittests/test_generate_mask_target_op.py]: bno is unnecessary

**Review by xingyuanbu — CHANGES_REQUESTED** [review]

File: python/paddle/fluid/tests/unittests/test_generate_mask_target_op.py (hunk 1 above)
Comment (jerrywgz) [reply python/paddle/fluid/tests/unittests/test_generate_mask_target_op.py]: Done

File: python/paddle/fluid/tests/unittests/test_generate_mask_target_op.py (hunk 2 above)
Comment (jerrywgz) [reply python/paddle/fluid/tests/unittests/test_generate_mask_target_op.py]: Removed

File: python/paddle/fluid/tests/unittests/test_generate_mask_target_op.py [hunk 3]
```diff
@@ -0,0 +1,272 @@
...
+    """Convert a list of masks into an array of tight bounding boxes."""
+    boxes_from_masks = np.zeros((len(masks), 4), dtype=np.float32)
+    for i in range(len(masks)):
+        mask = masks[i]
+        mask_loc = np.where(mask > 0)
+        xmin = np.min(mask_loc[1])
+        xmax = np.max(mask_loc[1])
+        ymin = np.min(mask_loc[0])
+        ymax = np.max(mask_loc[0])
+        boxes_from_masks[i, :] = np.array([xmin, ymax, xmin, ymax],\
```
Comment (xingyuanbu) [inline python/paddle/fluid/tests/unittests/test_generate_mask_target_op.py]: xmin, ymax, xmin, ymax, typo?

**Review by xingyuanbu — CHANGES_REQUESTED** [review]

File: python/paddle/fluid/tests/unittests/test_generate_mask_target_op.py (hunk 3 above)
Comment (jerrywgz) [reply python/paddle/fluid/tests/unittests/test_generate_mask_target_op.py]: Fixed

**Pull request closed without merging**